sys/xtimer/xtimer_core.c

```diff
@@ -176,42 +176,30 @@ static void _add_timer_to_list(xtimer_t **list_head, xtimer_t *timer)
 /**
  * @brief remove a timer from an ordered list of timers
  */
-static int _remove_timer_from_list(xtimer_t **list_head, xtimer_t *timer)
+static void _remove_timer_from_list(xtimer_t **list_head, xtimer_t *timer)
 {
     while (*list_head) {
         if (*list_head == timer) {
             *list_head = timer->next;
-            timer->offset = 0;
-            timer->long_offset = 0;
-            timer->start_time = 0;
             timer->next = NULL;
-            return 1;
```

Comment:
If you don't need the value anymore that fine, but the return should definitely stay here ;)

Reply:
Wow. That is a great mistake. Thanks!